docs/checks/internal/validate/main.go

```diff
@@ -14,235 +14,21 @@
 package main
 
 import (
-	"errors"
 	"fmt"
-	"os"
-	"path"
-	"regexp"
 	"strings"
 
-	"github.com/google/go-cmp/cmp"
-	"github.com/google/go-cmp/cmp/cmpopts"
-
 	"github.com/ossf/scorecard/v4/checks"
 	docs "github.com/ossf/scorecard/v4/docs/checks"
 )
 
-var (
-	allowedRisks     = map[string]bool{"Critical": true, "High": true, "Medium": true, "Low": true}
-	allowedRepoTypes = map[string]bool{"GitHub": true, "local": true}
-	supportedAPIs    = map[string][]string{
-		// InitRepo is supported for local repos in general. However, in the context of checks,
-		// this is only used to look up remote data, e.g. in Fuzzing check.
-		// So we only have "GitHub" supported.
```

Comment:
I think it's dangerous to remove some sanity checks. This code already caught one problem when someone updated a check and it would have changed the supported checks. I'm pretty sure we will update some checks; forget to update the `registerCheck()` and it will break the `--local` feature, and later the actions running on pull requests (not quite there yet though :-)) Today we don't have unit tests for the `--repo` option, which will make it hard to catch.

What can we do to add sanity checks?

Reply:
#435 would add e2e tests for `--local` option?

Reply:
in the pipeline, yes.

Reply:
We can consider prioritizing it if that addresses your concern here. I need this to unblock myself for `commit-based` Scorecard work. Would prefer getting this PR in and in parallel working to get the e2e tests for this.

Reply:
I think this should work. Let's try to avoid releasing a scorecard action before these checks are back in.

Reply:
SGTM.